homeassistant/components/homekit_controller/config_flow.py

```diff
@@ -270,7 +271,29 @@ async def async_step_zeroconf(self, discovery_info):
         # invalid. Remove it automatically.
         existing = find_existing_host(self.hass, hkid)
         if not paired and existing:
-            await self.hass.config_entries.async_remove(existing.entry_id)
+            if self.controller is None:
+                await self._async_setup_controller()
+
+            pairing = self.controller.load_pairing(
+                existing.data["AccessoryPairingID"], dict(existing.data)
+            )
+            try:
+                await pairing.list_accessories_and_characteristics()
+                _LOGGER.debug(
```

Comment:
Please only wrap the line that can raise in the try... except block. Use an `else:` block if needed.